### ISSUE TITLE
Indexing is successful even if the types are not supported/unexpected

### ISSUE THREAD

**Issue by yeikel** [zouzias/spark-lucenerdd#165]
**Describe the bug**
If indexing process is not successful( Example : Unsupported types) , logs show that the document was indexed successfully. 

**To Reproduce**

```
    val entities = Seq(
      ("123",Seq("Peter", "Johan"),"123"),
      ("125",Seq("Peter"),"123"),
      ("1234",Seq("Marc", "Johan"),"210")
    ).toDF("id","names","address").coalesce(1)

    val entitiesRDD = LuceneRDD(entities)

   entitiesRDD.count
```

`names `is not supported because it is a multivalued entry and the return document from the indexing is an empty document. 

![image](https://user-images.githubusercontent.com/8935151/55216104-42f75c00-51d2-11e9-98aa-768c83d74e61.png)

```
19/03/29 03:26:08 INFO LuceneRDDPartition: Index disk location C:\Users\user\AppData\Local\Temp\
19/03/29 03:26:08 INFO LuceneRDDPartition: Config parameter lucenerdd.index.store.mode is set to 'disk'
19/03/29 03:26:08 INFO LuceneRDDPartition: Lucene index will be storage in disk
19/03/29 03:26:08 INFO LuceneRDDPartition: Index disk location C:\Users\user\AppData\Local\Temp\
19/03/29 03:26:08 INFO LuceneRDDPartition: [partId=0] Partition is created...
19/03/29 03:26:08 INFO LuceneRDDPartition: [partId=0]Indexing process initiated at 2019-03-29T03:26:08.170-04:00...
19/03/29 03:26:08 INFO LuceneRDDPartition: [partId=0]Indexing process completed at 2019-03-29T03:26:08.755-04:00...
19/03/29 03:26:08 INFO LuceneRDDPartition: [partId=0]Indexing process took 0 seconds...
19/03/29 03:26:08 INFO LuceneRDDPartition: [partId=0]Indexed 1 documents
```

**Expected behavior**

I believe indexing should fail or at least give a warning

This may be related to #115

**Comment by yeikel** [comment]
Perhaps this [method ](https://github.com/zouzias/spark-lucenerdd/blob/master/src/main/scala/org/zouzias/spark/lucenerdd/package.scala#L118)


Should throw an exception or a warning for _ instead of empty doc. It is very confusing IMHO.

**Comment by zouzias** [comment]
I agree. I need to fix this issue asap and probably throw an Implicit convension not found or another exception.

**Comment by zouzias** [comment]
To be fixed in 0.3.7
